File: Zeze/Transaction/RedoAndReleaseLockException.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Zeze.Transaction
{
    public sealed class RedoAndReleaseLockException : Exception
    {
        public TableKey TableKey { get; }
        public long GlobalSerialId { get; }

        internal RedoAndReleaseLockException(TableKey tkey, long serialId, string msg)
            : base(msg)
        {
            TableKey = tkey;
            GlobalSerialId = serialId;
        }
    }
}

File: ZezeJava/src/Zeze/Transaction/RedoAndReleaseLockException.java
package Zeze.Transaction;

public final class RedoAndReleaseLockException extends Error {
	/**
	 * 
	 */
	private static final long serialVersionUID = -7674037466857621440L;

	public TableKey TableKey;
	public long GlobalSerialId;

	public RedoAndReleaseLockException() {
	}

	public RedoAndReleaseLockException(TableKey tkey, long serialId, String msg) {
		super(msg);
		TableKey = tkey;
		GlobalSerialId = serialId;
	}
}